### ISSUE TITLE
Feature: [unified-signatures] Option to not report when signature has different JSDoc

### ISSUE THREAD

**Issue by nicolo-ribaudo** [typescript-eslint/typescript-eslint#10520]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.4.5&showAST=es&fileType=.tsx&code=FAEwpgxgNghgTmABAMwK4DsIBcCWB7dFACgA8AuRdVAWwCMw4BKCjAa3TwHd0BuYAegBUgxAAFwABwQQYWMCESD%2BicNHhI0mXAWLlEtPHihgY6ZojYdufYMiIBGRnztY4qME%2BBA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6WJygM0sQBNaySgHMmAQ3yxoKdGACM4MAF8QSoA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
declare function f(x: number): unknown;
/** @deprecated */ declare function f(x: boolean): unknown;

f(1);
f(true);
```

### ESLint Config

```javascript
{
  "rules": {
    "@typescript-eslint/unified-signatures": 1
  }
}
```

### Expected Result

It should not warn, because the two signatures cannot be unified. Note that the call `f(true)` is striked as "deprecated", and it would not be possible for it to happen if the two function declarations were merged into a single one.

### Actual Result

It reports
> These overloads can be combined into one signature taking `number | boolean`.

**Comment by kirkwaiblinger** [comment]
Hi @nicolo-ribaudo ! 

I can see the argument, but I'm not sure it will make sense for us to support this. We generally don't touch JSDocs. The sort-of exception to this is no-deprecated, which works via TS's type information, rather than us interacting with JSDocs directly, but that isn't available to us here, since unified-signatures isn't type-aware.

My intuition is to think of this as a bandwidth/scope limitation that is best solved with an eslint-disable comment. Curious for thoughts from @typescript-eslint/triage-team.

**Comment by JoshuaKGoldberg** [comment]
Blurgh. If unified-signatures was type-aware I think this would be easily something to accept, no complications.

But, since the JSDocs are all in-place, maybe we can fake it? Just go purely off syntax by getting and parsing the JSDoc comments for the nodes themselves? Performance shouldn't be impacted significantly given that it's only one node per overload.

**Comment by kirkwaiblinger** [comment]
If we entertain this, then I'd ask whether we should generally consider signatures that have different JSDocs 🤔....

[Playground](https://typescript-eslint.io/play/#ts=5.7.2&fileType=.ts&code=PQKhCgAIUgVALAlgZ0sxBzAdgQwC4CuATgKaQAmA9iaspQLYl5JYYB0UIw45JAxgBscpSADMCWPnkSUsYgBQAPAFyQsBegCMSRAJSqAbpUTkA3OHCgI0OElqZchEVRpoGTFhkgkByEh2huXkFhMnFJaVkFFTQ8IkRWfUgjE3MgA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6WJygM0sQBNaySgHMmAQ3yxoKdFETRoAe2iRwYAL4h1QA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false)

```ts
/**
 * This signature does something.
 */
declare function f(x: number): void;

/**
 * This signature does something else.
 */
declare function f(x: string): void;
```

**Comment by bradzacher** [comment]
Yeah I don't think we really want to go into the realm of "partially supporting specific cases of JSDoc and ignoring all other comment contents" cos it does create an inconsistent experience.

I think either we treat JSDoc comments as a marker of uniqueness (behind a default off flag) or we continue to ignore them.

**Comment by JoshuaKGoldberg** [comment]
👍. Retracting any suggestion of comment-specific parsing.

As for this specific issue, since TypeScript knows to show the deprecated `f(true)` as deprecated but not `f(1)`, we should too. Accepting PRs.

**Comment by JoshuaKGoldberg** [comment]
👍 agreed. Marking as accepting PRs for a default-off flag as a feature request for the rule.
